software/auth/auth.go

```diff
@@ -127,6 +127,20 @@ func registryAuth(client houston.ClientInterface, out io.Writer) error {
 	return nil
 }
 
+func getWorkspaces(client houston.ClientInterface, interactive bool) ([]houston.Workspace, error) {
+	var workspaces []houston.Workspace
+	var err error
+
+	if interactive {
+		workspacePageSize := 2
```

Comment:
why not use the `PageSize` value from `config.CFG.PageSize` or else `defaultPageSize`?
Reference: https://github.com/astronomer/astro-cli/blob/main/software/auth/auth.go#L135

Reply:
Since we need to identify if the number of workspaces returned 1 or more, calling workspace pagination with page size 2 if the interactive flag is set to true

Reply:
That's not entirely true, because the list of workspaces returned from this method is used to switch to `last_used_workspace`, and by just pulling two workspaces, that functionality will almost always be broken in case interactive is set to true.

So from a UX point of view, if interactive is set to true, then whenever a user tries to log in, they would be asked to select a particular workspace every single time, and not automatically set the last used workspace from the previous login as the current workspace.

We should document this in our release note if we are going with this approach, and also add comments in the code around the logic of choosing the page size as 2 for future reference.

Might want to check once on this functional change with @cmarteepants if not already done.

Reply:
The trade-off (selecting the workspace every single time a user logs in) is too expensive from a UX standpoint. I'm not following how the change helps solve the initial problem. Is there another approach we can use to solve the problem, while retaining the login UX?

Reply:
As discussed, I'll change the implementation of `switchToLastUsedWorkspace` to use `getWorkspace` API to identify if the user still has access/available  `last_used_workspace ` workspace. That way will have the same user experience.